test/e2e/apps/statefulset.go

```diff
@@ -1345,6 +1346,43 @@ var _ = SIGDescribe("StatefulSet", func() {
 			framework.ExpectNoError(err)
 		})
 	})
+
+	// add for issue: https://github.com/kubernetes/kubernetes/issues/108837
+	ginkgo.It("Be sure to update the status even if the new pod creation fails", func() {
```

Comment:
Can you test this in integration tests instead of e2e? just because they are faster to run.

Reply:
I have updated the code to test it in integration tests. Please take a look. Thanks.